### PR TITLE
Fixing the Simulated and Adding a new Fn TransactionByHashWithBlockNum

### PR DESCRIPTION
**Pull request by piyuvyas**

### CODE REVIEW

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -188,6 +188,28 @@ func (tx *rpcTransaction) UnmarshalJSON(msg []byte) error {
 	return json.Unmarshal(msg, &tx.txExtraInfo)
 }
 
+// TransactionByHash returns the transaction with the given hash.
+func (ec *Client) TransactionByHashWithBlockNum(ctx context.Context, hash common.Hash) (tx *types.Transaction, blockHexString string, err error) {
+	var json *rpcTransaction
+	err = ec.c.CallContext(ctx, &json, "eth_getTransactionByHash", hash)
```
Comment (bmperrea) [inline ethclient/ethclient.go:197]: `&json` seems to be a pointer to a pointer

File: ethclient/ethclient.go [hunk 2]
```diff
@@ -188,6 +188,28 @@ func (tx *rpcTransaction) UnmarshalJSON(msg []byte) error {
...
+	if err != nil {
```
Comment (bmperrea) [inline ethclient/ethclient.go:198]: can you comment on where the meat of this code was copied from?

File: ethclient/ethclient.go (hunk 1 above)
Comment (piyuvyas) [reply ethclient/ethclient.go:197]: The code is exactly copied from the function TransactionByHash and it seems to be using the same there as well.  If you see the underlying callContext you can see its using a chan of type jsonResponse pointer and then using that Add to of the pointer to unmarshall the response from rpc

File: ethclient/ethclient.go (hunk 2 above)
Comment (piyuvyas) [reply ethclient/ethclient.go:198]: Added

**Review by bmperrea — APPROVED** [review]

**Pull request merged**